src/Bicep.LangServer/Deploy/DeploymentHelper.cs

```diff
@@ -0,0 +1,114 @@
+using Azure.ResourceManager.Resources.Models;
+using Bicep.Core;
+
+namespace Bicep.LanguageServer.Deploy
+{
+    public class DeploymentHelper
+    {
+        public static async Task<string> CreateDeployment(
+            IDeploymentCollectionProvider deploymentCollectionProvider,
+            ArmClient armClient,
+            string template,
+            string parameterFilePath,
+            string id,
+            string scope,
+            string location)
+        {
+            if ((scope == LanguageConstants.TargetScopeTypeSubscription ||
+                scope == LanguageConstants.TargetScopeTypeManagementGroup) &&
+                string.IsNullOrWhiteSpace(location))
+            {
+                return LangServerResources.MissingLocationDeploymentFailedMessage;
+            }
+
+            DeploymentCollection? deploymentCollection;
+            var resourceIdentifier = new ResourceIdentifier(id);
+
+            if (scope == LanguageConstants.TargetScopeTypeResourceGroup ||
+                scope == LanguageConstants.TargetScopeTypeSubscription ||
+                scope == LanguageConstants.TargetScopeTypeManagementGroup)
+            {
+                try
+                {
+                    deploymentCollection = deploymentCollectionProvider.GetDeploymentCollection(armClient, resourceIdentifier, scope);
+                }
+                catch (Exception e)
+                {
+                    return string.Format(LangServerResources.DeploymentFailedWithExceptionMessage, e.Message);
+                }
+            }
+            else
+            {
+                return LangServerResources.UnsupportedScopeDeploymentFailedMessage;
+            }
+
+            if (deploymentCollection is not null)
+            {
+                JsonElement parameters;
+
+                if (string.IsNullOrWhiteSpace(parameterFilePath))
+                {
+                    parameters = JsonDocument.Parse("{}").RootElement;
+                }
+                else
+                {
+                    try
+                    {
+                        string text = File.ReadAllText(parameterFilePath);
+                        parameters = JsonDocument.Parse(text).RootElement;
+                    }
+                    catch(Exception e)
+                    {
+                        return string.Format(LangServerResources.InvalidParameterFileDeploymentFailedMessage, e.Message);
+                    }
+                }
+
+                var deploymentProperties = new DeploymentProperties(DeploymentMode.Incremental)
+                {
+                    Template = JsonDocument.Parse(template).RootElement,
+                    Parameters = parameters
+                };
+                var input = new DeploymentInput(deploymentProperties)
+                {
+                    Location = location,
+                };
+
+                string deployment = "deployment_" + DateTime.UtcNow.ToString("yyyyMMddHmmffff");
```

Comment:
The date and time string doesn't contain seconds. Could we replace `ffff` with `ss` and change `H` to `HH`?

Reply:
Also, should we allow users to provide custom deployment names?

Reply:
@ucheNkadiCode , @StephenWeatherford  and I discussed at one point about not letting custom deployment names. I don't recollect the reason.

@ucheNkadiCode, is it something we would want to support in future?

Reply:
I can't imagine a common use for it.  If people ask for it, we can consider it, until then keep it simple.

Reply:
I like the idea of having seconds contained in the date and time string. I'd like you to include @shenglol's suggestion here @bhsubra.

As for the deployment name, I agree with Stephen, there isn't a largely common use for it and folks haven't mentioned it yet as a feature. I imagine this won't be a difficult addition if we ever do add it. Keep it simple for now.

Reply:
I am resolving this conversation. Let me know if there are any concerns here. Thanks!